crates/gitbutler-serde/src/lib.rs

```diff
@@ -3,6 +3,28 @@ use serde::Serialize;
 mod bstring;
 pub use bstring::BStringForFrontend;
 
+pub mod bstring_lossy {
+    use bstr::{BString, ByteSlice};
+    use serde::{Deserialize, Deserializer, Serialize};
+
+    /// serialize an object ID as hex-string.
+    pub fn serialize<S>(v: &BString, s: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        v.to_str_lossy().serialize(s)
+    }
+
+    /// deserialize an object ID from hex-string.
+    pub fn deserialize<'de, D>(d: D) -> Result<BString, D::Error>
```

Comment:
Maybe we dont want a lossy deserialization?

Reply:
I kind of needed this code so https://github.com/gitbutlerapp/gitbutler/pull/6045

Reply:
A great catch!After extra-validation I could conclude that in order to do it correctly for TreeChanges, we'd have to keep the original byte buffer around to avoid the frontend having to deal with it. Tauri would then once again have a duplicate type with the extra byte buffer, just so that the plumbing types isn't 'polluted' by it.